x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/jwt/JwtHeaderValidator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.authc.jwt;
+
+import com.nimbusds.jose.JOSEObjectType;
+import com.nimbusds.jose.JWSAlgorithm;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.proc.BadJOSEException;
+import com.nimbusds.jose.proc.DefaultJOSEObjectTypeVerifier;
+import com.nimbusds.jose.proc.JOSEObjectTypeVerifier;
+import com.nimbusds.jose.proc.SecurityContext;
+import com.nimbusds.jwt.SignedJWT;
+
+import org.elasticsearch.ElasticsearchSecurityException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.rest.RestStatus;
+
+import java.util.List;
+
+public class JwtHeaderValidator implements JwtClaimValidator {
```

Comment:
Yeah I was obsessed with one Validator interface. I think it is not necessary. I have made `JwtHeaderValidator` a class of its own. Because of that, both the `JwtClaimValidtor#validate` and `JwtHeaderValidator#validate` methods can be simplified to not take the entire SignedJWT but only the relevant segment of it.